design-proposals/passt-feature-gate-iface-api-deprecation.md

```diff
@@ -0,0 +1,62 @@
+## Overview
+Following the emerging [Kubevirt Network Binding Plugin API](https://docs.google.com/document/d/13fLaqZuM8El_CT5XzxPv-1epKsB1-Spv_jtXveEkkdQ/edit#heading=h.z27rudfpwy44)
+and introduction of Passt network binding plugin [2](https://github.com/kubevirt/kubevirt/pull/10425) [3](https://github.com/kubevirt/kubevirt/pull/10621), 
+it's now possible to remove Passt network binding implementation from Kubevirt core.
+
+This proposal outlines the replacement plan for the existing Passt implementation and API.
+
+# Motivation
+Removing Passt network binding implementation from Kubevirt core enable reduce the maintenance burden.
+Having Passt binding code outside Kubevirt's core may ease contribution process as there is no need to be familiar 
+with [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt) core code.
+
+## Goals
+- Reduce maintenance of Passt network binding in Kubevirt core.
+  
+## Non Goals
+
+## User Stories
+
+## Repos
+[kubevirt/kubevirt](https://github.com/kubevirt/kubevirt):
+- [Passt binding plugin hook sidecar](https://github.com/kubevirt/kubevirt/tree/main/cmd/sidecars/network-passt-binding)
+- [Passt binding plugin CNI](https://github.com/kubevirt/kubevirt/tree/main/cmd/cniplugins/passt-binding)
+
+# Design
+Replace Kubevirt core Passt network binding implementation and interface API with Passt binding plugin.
+Users should be aware of the coming changes and be prepared.
+
+The proposed plan for replacing Kubevirt core Passt binding implementation and API is as follows:
+
+1. Announce the coming changes for Passt binding in kubevirt-dev mailing list.
+   This will give users some time to prepare for changes in Kubevirt next release v1.2.0. 
+
+2. Kubevirt next release v1.1.1 should have the following changes:
+   Raise warning when Passt feature gate is enabled saying its deprecated
+   Raise warning when Passt interface API is being used saying its deprecated and Passt binding plugin should be used instead.
+ 
+3. Kubevirt next release v1.2.0 should have the following changes:
+   1. [Passt interface API](http://kubevirt.io/api-reference/main/definitions.html#_v1_interfacepasst) 
+      set to be no-op.
+      When specified, the default network binding will be set instead (bridge/masquerade).
```

Comment:
I admit this is a hard one.

On one hand, it will be nicer not to break existing workloads.
On the other hand, it is indeed problematic to fallback to something that is not the desired spec. Although, the system itself is fine with having a diff between the desired and the actual, as long as it is announced/exposed.

The first option, to block creation, is the easiest to implement.
It is also the easiest to fix if we get feedback that this is causing too much pain.
I guess we can start with it and see if there is other feedback about it.